chains/evm/contracts/pools/FastTransferTokenPoolAbstract.sol

```diff
@@ -0,0 +1,431 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+// Local interfaces
+
+import {IAny2EVMMessageReceiver} from "../interfaces/IAny2EVMMessageReceiver.sol";
+import {IFastTransferPool} from "../interfaces/IFastTransferPool.sol";
+import {IRouterClient} from "../interfaces/IRouterClient.sol";
+
+import {IRMN} from "../interfaces/IRMN.sol";
+import {ITypeAndVersion} from "@chainlink/contracts/src/v0.8/shared/interfaces/ITypeAndVersion.sol";
+
+import {CCIPReceiver} from "../applications/CCIPReceiver.sol";
+import {Client} from "../libraries/Client.sol";
+import {TokenPool} from "./TokenPool.sol";
+
+import {IERC20} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC165} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v4.8.3/contracts/utils/introspection/IERC165.sol";
+
+/// @title Abstract Fast-Transfer Pool
+/// @notice Base contract for fast-transfer pools that provides common functionality
+/// for quoting, fill-tracking, and CCIP send helpers.
+abstract contract FastTransferTokenPoolAbstract is TokenPool, CCIPReceiver, ITypeAndVersion, IFastTransferPool {
+  using SafeERC20 for IERC20;
+
+  error WhitelistNotEnabled();
+  error InvalidLaneConfig();
+  error FillerNotWhitelisted(uint64 remoteChainSelector, address filler);
+
+  event LaneUpdated(
+    uint64 indexed dst,
+    uint16 bps,
+    uint256 fillAmountMaxPerRequest,
+    bytes destinationPool,
+    address[] addFillers,
+    address[] removeFillers
+  );
+  event FillerAllowListUpdated(uint64 indexed dst, address[] addFillers, address[] removeFillers);
+  event DestinationPoolUpdated(uint64 indexed dst, address destinationPool);
+
+  struct LaneConfig {
+    uint256 fillAmountMaxPerRequest; //    max amount that can be filled per request
+    uint16 bpsFastFee; // ─────────────╮ 0-10_000
+    bool fillerAllowlistEnabled; // ───╯ whitelist for fillers
+    bytes destinationPool; // destination pool address
+    mapping(address filler => bool isAllowed) fillerAllowList; // whitelist of fillers
+  }
+
+  struct LaneConfigView {
+    uint256 fillAmountMaxPerRequest; //    max amount that can be filled per request
+    uint16 bpsFastFee; // ────────────╮ 0-10_000
+    bool fillerAllowlistEnabled; // ──╯ whitelist for fillers
+    bytes destinationPool; // destination pool address
+  }
+
+  struct LaneConfigArgs {
```

Comment:
can this be changed to LaneConfigUpdateArgs to be more descriptive please, that way it isn't confused with LaneConfig and known to be for updates similar to `struct ChainUpdate`